### PR TITLE
Editor | Implement Add User form

### PR DESCRIPTION
**Pull request by mddilley**
Closes part of https://github.com/cityofaustin/atd-data-tech/issues/4409

This PR updates the editor with a form to add a user. Basic Yup validations have been added to require fields before submission. This will be extended in the future. The form has been tested to create a user after pulling the latest updates from `main` branch. The edit and delete features are coming in a separate PR to keep this from getting too large.

### Add User Form
Screen Shot 2020-11-17 at 6 29 44 PM

### Button link to Add User Form
Screen Shot 2020-11-17 at 6 35 49 PM

### New test user in Cognito Dashboard
Screen Shot 2020-11-17 at 6 30 11 PM

### CODE REVIEW

File: moped-editor/src/views/staff/StaffForm.js [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+  first_name: yup.string().required(),
+  last_name: yup.string().required(),
+  title: yup.string().required(),
+  workgroup: yup.string().required(),
+  workgroup_id: yup.string().required(),
+  email: yup.string().required(),
+  password: yup.string().required(),
+  roles: yup.string().required(),
+  status_id: yup.string().required(),
+});
```
Comment (sergiogcx) [inline moped-editor/src/views/staff/StaffForm.js:72]: Very cool way to validate data! :)

File: moped-editor/src/views/staff/StaffForm.js [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+
+const fieldParsers = {
+  status_id: id => parseInt(id),
+  workgroup_id: id => parseInt(id),
+  roles: role => [role],
+};
+
+const StaffForm = ({ editFormData = null }) => {
+  const classes = useStyles();
+  const [userApiResult, userApiLoading, requestApi] = useUserApi();
```
Comment (sergiogcx) [inline moped-editor/src/views/staff/StaffForm.js:82]: Love the custom Hook! 👍

**Review by sergiogcx — APPROVED** [review]
It works! I was able to create a user, my only question is about feedback. I had forgotten to select the workgroup the user belongs to and I did not see anything in the console or in the UI that would remind me. 

Otherwise, thank you for showing me how to use custom hooks to integrate with an API and how to use YUP :)

File: moped-editor/src/views/staff/StaffForm.js (hunk 1 above)
Comment (mddilley) [reply moped-editor/src/views/staff/StaffForm.js:72]: Thanks, Yup plugs right into React Hook Form and makes it nice to add. 👍 Also, great catch on the validation helper text for the workgroup field. I added that in the next branch with the edit/delete features. 🙏

**Pull request merged**